scipy/io/mmio.py

```diff
@@ -281,20 +286,44 @@ def _get_symmetry(a):
         m,n = a.shape
         if m != n:
             return MMFile.SYMMETRY_GENERAL
-        issymm = 1
-        isskew = 1
+        issymm = True
+        isskew = True
         isherm = a.dtype.char in 'FD'
-        for j in range(n):
-            for i in range(j+1,n):
-                aij,aji = a[i][j],a[j][i]
-                if issymm and aij != aji:
-                    issymm = 0
-                if isskew and aij != -aji:
-                    isskew = 0
-                if isherm and aij != conj(aji):
-                    isherm = 0
-                if not (issymm or isskew or isherm):
-                    break
+        
+        if isinstance(a,spmatrix):
+            # check if number of nonzero entrys of lower and upper triangle matrix are equal
+            (row, col) = a.nonzero()
+            if (row < col).sum() != (row > col).sum():
+                return MMFile.SYMMETRY_GENERAL
+            
```

Comment:
I think my algorithm is correct if `argsort` is "stable" - so the argsort call would need to be `sort_cols = np.argsort(cols, kind='merge')`.  Am I wrong in that case?  Have you got a counter-example?

I agree it's not clear whether the effort is worth it.  To get a good answer I guess we'd have to estimate the proportion of matrices that would reach this routine that did have the same number of entries below and above the diagonal, but where the entries were not compatible, and check this for different sizes.   The problem is also that the entry-by-entry check for equality below is likely to be horribly slow for large matrices as done in Python, whereas the work for this sorting check is done at the C level.